firebase-common-internal/src/commonMain/kotlin/dev/gitlive/firebase/internal/EncodedObject.kt

```diff
@@ -0,0 +1,27 @@
+package dev.gitlive.firebase.internal
+
+import kotlin.jvm.JvmInline
+
+/**
+ * Platform specific object for storing encoded data that can be used for methods that explicitly require an object.
+ * This is essentially a [Map] of [String] and [Any]? (as represented by [raw]) but since [encode] gives a platform specific value, this method wraps that.
+ */
+sealed interface EncodedObject {
+    val raw: Map<String, Any?>
+}
```

Comment:
whats the purpose of still exposing this to clients?

Reply:
Figured it would be useful. Moved to an internal get method now (cant be value due to JS weirdness)